utils/extract.sh

```diff
@@ -0,0 +1,95 @@
+#!/usr/bin/env bash
+
+# usage: extract.sh filename [destination]
+#   destination - directory to extract to. Will be created if non-existent. Leaving empty
+#     will result in extraction to the current working directory.
+
+# Requires:
+#   - tar and gzip
+#   - zip
+#   - 7z
+
+echoerr() {
+    echo "extract.sh:" "$@" >&2
+}
+
+
+# Preliminary checks
+if ! [ -f "$1" ]; then
+    echoerr "'$1' does not exist, aborting"
+    exit 1
+fi
+
+extension="${1#*.}"
+
+case "$extension" in
+    "zip")
+        cmds_needed=( "unzip" )
+        ;;
+    "tar.gz")
+        cmds_needed=( "tar" "gzip" )
+        ;;
+    "7z")
+        cmds_needed=( "7z" )
+        ;;
+    *)
+        echoerr "cannot extract '$1' - unsupported archive type"
+        exit 1
+        ;;
+esac
+
+for cmd in "${cmds_needed[@]}"; do
+    if ! command -v $cmd >& /dev/null; then
+        echoerr "command '$cmd' not found but is required for file '$1', aborting"
+        exit 1
+    fi
+done
+
+
+# Now we can begin the extraction
+extractdir="$2"
+if [ "$extractdir" == "" ]; then
+    extractdir="."
```

Comment:
I don't see much of a use case for this, I would prefer for the script to exit with an error. Would also remove the need for the check at [ln 67](https://github.com/rockerbacon/lutris-skyrimse-installers/blob/814f5c38e4d568313454ffd0b668dc658d0324f7/utils/extract.sh#L67).

Reply:
OK, cool. I'll make the second argument required then (and reindent with tabs).

Reply:
fixed. In regard to automated checks, you may want to consider running all bash scripts in PRs through `shellcheck`, which is a very useful utility for this kind of work :)

Reply:
I've used shellcheck in other projects and I'll consider adding it to this one later. But shellcheck doesn't do any style checking, for that you need something like [shfmt](https://github.com/mvdan/sh).